### PR TITLE
PUD-1465: Use correct filenames for files that are generated by the api the first time (Reasons for recall & Revocation Order)

### PR DESCRIPTION
**Pull request by philsegal**

### CODE REVIEW

File: src/main/kotlin/uk/gov/justice/digital/hmpps/managerecallsapi/config/OpenApiConfiguration.kt [hunk 1]
```diff
@@ -41,32 +42,32 @@ class OpenApiConfiguration(buildProperties: BuildProperties) {
...
         .replaceWithClass(PoliceForceName::class.java, String::class.java)
-        .replaceWithClass(FieldName::class.java, String::class.java)
-        .replaceWithClass(FieldPath::class.java, String::class.java)
-        .replaceWithClass(ColumnName::class.java, String::class.java)
+        .replaceWithClass(PrisonId::class.java, String::class.java)
+        .replaceWithClass(PrisonName::class.java, String::class.java)
+        .replaceWithClass(RecallId::class.java, UUID::class.java)
+        .replaceWithClass(RescindRecordId::class.java, UUID::class.java)
+        .replaceWithClass(UserId::class.java, UUID::class.java)
+        .replaceWithClass(WarrantReferenceNumber::class.java, String::class.java)
```
Comment (philsegal) [inline src/main/kotlin/uk/gov/justice/digital/hmpps/managerecallsapi/config/OpenApiConfiguration.kt:70]: Alphabetised (and removed duplicate NomsNumber)

File: src/main/kotlin/uk/gov/justice/digital/hmpps/managerecallsapi/config/RestConfiguration.kt [hunk 2]
```diff
@@ -72,30 +73,31 @@ object ManageRecallsApiJackson : ConfigurableJackson(
...
+  text(::WarrantReferenceNumber)
   text(StringBiDiMappings.uuid().map(::DocumentId, DocumentId::value))
-  text(StringBiDiMappings.uuid().map(::UserId, UserId::value))
-  text(StringBiDiMappings.uuid().map(::RescindRecordId, RescindRecordId::value))
+  text(StringBiDiMappings.uuid().map(::LastKnownAddressId, LastKnownAddressId::value))
+  text(StringBiDiMappings.uuid().map(::MissingDocumentsRecordId, MissingDocumentsRecordId::value))
   text(StringBiDiMappings.uuid().map(::NoteId, NoteId::value))
+  text(StringBiDiMappings.uuid().map(::RecallId, RecallId::value))
+  text(StringBiDiMappings.uuid().map(::RescindRecordId, RescindRecordId::value))
+  text(StringBiDiMappings.uuid().map(::UserId, UserId::value))
```
Comment (philsegal) [inline src/main/kotlin/uk/gov/justice/digital/hmpps/managerecallsapi/config/RestConfiguration.kt:100]: Alphabetised & added FileName

File: src/main/kotlin/uk/gov/justice/digital/hmpps/managerecallsapi/db/Converters.kt [hunk 3]
```diff
@@ -13,16 +14,17 @@ import uk.gov.justice.digital.hmpps.managerecallsapi.domain.PrisonId
...
-class PhoneNumberJpaConverter : CustomJpaConverter<PhoneNumber, String>({ it.value }, ::PhoneNumber)
+class MiddleNamesJpaConverter : CustomJpaConverter<MiddleNames?, String?>({ it?.value }, { it?.let { MiddleNames(it) } })
 class NomsNumberJpaConverter : CustomJpaConverter<NomsNumber, String>({ it.value }, ::NomsNumber)
-class CroNumberJpaConverter : CustomJpaConverter<CroNumber, String>({ it.value }, ::CroNumber)
-class PrisonIdJpaConverter : CustomJpaConverter<PrisonId?, String?>({ it?.value }, { it?.let { PrisonId(it) } })
-class CourtIdJpaConverter : CustomJpaConverter<CourtId?, String?>({ it?.value }, { it?.let { CourtId(it) } })
+class PhoneNumberJpaConverter : CustomJpaConverter<PhoneNumber, String>({ it.value }, ::PhoneNumber)
 class PoliceForceIdJpaConverter : CustomJpaConverter<PoliceForceId?, String?>({ it?.value }, { it?.let { PoliceForceId(it) } })
+class PrisonIdJpaConverter : CustomJpaConverter<PrisonId?, String?>({ it?.value }, { it?.let { PrisonId(it) } })
 class WarrantReferenceNumberJpaConverter : CustomJpaConverter<WarrantReferenceNumber?, String?>({ it?.value }, { it?.let { WarrantReferenceNumber(it) } })
```
Comment (philsegal) [inline src/main/kotlin/uk/gov/justice/digital/hmpps/managerecallsapi/db/Converters.kt:28]: Alphabetised & added FileName

File: src/test/kotlin/uk/gov/justice/digital/hmpps/managerecallsapi/integration/documents/GotenbergComponentTestBase.kt [hunk 4]
```diff
@@ -120,14 +121,14 @@ abstract class GotenbergComponentTestBase : ComponentTestBase(useRealGotenbergSe
...
-      UploadDocumentRequest(DocumentCategory.PART_A_RECALL_REPORT, base64EncodedFileContents("/document/part_a.pdf"), "PART_A.pdf")
+      UploadDocumentRequest(DocumentCategory.PART_A_RECALL_REPORT, base64EncodedFileContents("/document/part_a.pdf"), FileName("PART_A.pdf"))
     )
   }
 
   protected fun uploadLicenceFor(recall: RecallResponse) {
     authenticatedClient.uploadDocument(
       recall.recallId,
-      UploadDocumentRequest(DocumentCategory.LICENCE, base64EncodedFileContents("/document/licence.pdf"), "PART_A.pdf")
+      UploadDocumentRequest(DocumentCategory.LICENCE, base64EncodedFileContents("/document/licence.pdf"), FileName("PART_A.pdf"))
```
Comment (AndyNewton) [inline src/test/kotlin/uk/gov/justice/digital/hmpps/managerecallsapi/integration/documents/GotenbergComponentTestBase.kt:131]: Tiny but this one is for a LICENSE so can we please not call it "PART_A.pdf"?  Ta.

File: src/test/kotlin/uk/gov/justice/digital/hmpps/managerecallsapi/service/DocumentServiceTest.kt [hunk 5]
```diff
@@ -358,7 +360,7 @@ internal class DocumentServiceTest {
   fun `update a document category to a versioned category also sets version to 1`() {
     val documentId = ::DocumentId.random()
     val now = OffsetDateTime.now()
-    val document = Document(documentId, recallId, UNCATEGORISED, "parta.pdf", null, null, now, createdByUserId)
+    val document = Document(documentId, recallId, UNCATEGORISED, FileName("parta.pdf"), null, null, now, createdByUserId)
```
Comment (AndyNewton) [inline src/test/kotlin/uk/gov/justice/digital/hmpps/managerecallsapi/service/DocumentServiceTest.kt:363]: Same thing - albeit was present before - and yes, makes no difference in practice.

**Review by AndyNewton — APPROVED** [review]
Take a peek but I think my comments don't need addressing TBH.

**Pull request merged**